pandas/tests/indexes/test_any_index.py

```diff
@@ -47,7 +50,17 @@ def test_mutability(index):
 
 def test_map_identity_mapping(index):
     # GH#12766
-    tm.assert_index_equal(index, index.map(lambda x: x))
+    result = index.map(lambda x: x)
+    if index._is_backward_compat_public_numeric_index:
```

Comment:
the point of this file is to have tests that dont have these branchings.  i guess OK if this is temporary